app/src/features/apiClient/screens/apiClient/components/views/components/request/components/KeyValueTable/KeyValueTableTypeCell.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useMemo, useEffect, useRef } from "react";
+import { Form, Dropdown, Menu, MenuProps, FormInstance } from "antd";
+import { DownOutlined } from "@ant-design/icons";
+import { KeyValuePair, ValueType } from "features/apiClient/types";
+import { getInferredType } from "features/apiClient/screens/apiClient/utils";
+
+const ALL_VALUE_TYPES: ValueType[] = Object.values(ValueType);
+
+interface KeyValueTypeCellProps {
+  record: KeyValuePair;
+  dataIndex: keyof KeyValuePair;
+  form: FormInstance;
+  onSave: () => Promise<void>;
+}
+
+const KeyValueTypeCell: React.FC<KeyValueTypeCellProps> = ({ record, dataIndex, form, onSave }) => {
+  const handleMenuClick: MenuProps["onClick"] = useCallback(
+    ({ key }) => {
+      const selectedValueType = key as ValueType;
+      form.setFieldsValue({ [dataIndex]: selectedValueType });
+      onSave();
+    },
+    [form, dataIndex, onSave]
+  );
+
+  const menuItems: MenuProps["items"] = useMemo(
+    () =>
+      ALL_VALUE_TYPES.map((type) => ({
+        key: type,
+        label: type,
+        className: "key-value-type-dropdown-item",
+      })),
+    []
+  );
+
+  const menu = <Menu items={menuItems} onClick={handleMenuClick} />;
+
+  const displayType = useMemo(() => {
+    const explicitType = record?.[dataIndex] as ValueType | undefined;
+    return explicitType || getInferredType(record.value);
+  }, [record, dataIndex]);
+
+  const initializedRef = useRef<Set<number>>(new Set());
+  useEffect(() => {
+    const currentSavedType = record?.[dataIndex];
+    if (!currentSavedType && !initializedRef.current.has(record.id)) {
+      form.setFieldsValue({ [dataIndex]: displayType });
+      onSave();
+      initializedRef.current.add(record.id);
+    }
+  }, [record, dataIndex, displayType, form, onSave]);
```

Comment:
we don't need to correctly infer for old entries. can just default to using the `string` type for this